src/writeckpt.cpp

```diff
@@ -171,16 +171,10 @@ mtcp_writememoryareas(int fd)
    *        restart() is heavily overloaded, but assuming it's only relevant
    *        to processinfo.cpp, we have it used in: ProcessInfo::restart()
    *        and 'case DMTCP_EVENT_RESTART' in processInfo_EventHook().
-   *
```

Comment:
And this is the related comment.  See my above inline comment.

Conceptually, we have non-local code.  Data structures must be maintained consistently between writeckpt.cpp and mtcp_restart.c  So, we need to document the consistency requirements in both places.  Otherwise, new developers have a hard time to understand the code.